sdk/iot/doc/readme.md

```diff
@@ -0,0 +1,294 @@
+
+### Device Provisioning Service
+
+![iot_provisioning_flow](https://www.plantuml.com/plantuml/svg/0/hLTHRziu37xNhz3RaiCo70xRW0L3CJSjq6xhDTTum81X41IBSOJOaYV9MRFX_lient6onedjP7q82fFyI7waAEgTTTouF4q8iGkfCcvuIl6R0_gJKFoxvD4YDZfNOgGJRTw-3SRZYmkAlnzlHq5uU_zllTeaVFWq2b7p8r-24c38_-YY08wy1ekqa2eklQm5awiFB1ZVFlbo5OG1kxzyVpqAwHPtHsad5ZQwlxgcPP3UupDwgwGZN_xwIWU32yf_CAaTounUIYkrmWXOb4XGZadK6z9963o4f46j2Ie6nt8BXYcU0cchs0Pr4uIWG_hvrTzlAC_TlAILPraY8-xGYgNV72fhfI1o5HsjiOj03g1vIkyp_vadT19CgDCBK3G61QRG7dRmSpHVossXRsMikCiDOCxUOP5o9xidwSnJ7ltpwZMf-TOof8nR-rqRGJhe7cKLKwimzwkv2MvJI2p199_QoCWfa8EG3Ura89RMOUyLN6tWcDEPZDXC9YWYvvdNPrpf7chRIg7AzzaHdLeV1YAGIzymT_-58Ktab5OMmjFtK4PFcf1R4f3bqqJQ-JfzSf9LVIQbof-4BCavH7-9i6ss0teRZZStma-Al1PqZpzTqb8gYZ7TOlhYnMDXWodADk9Ae62LQ2w6-z1qPdHAfmRdZw6zAD6veDhGQn53lt1xXhlMnJTWCwqi_167dgSXLThHNGXCEC4AODys-DWBgXtfEYidD_ELpYrfbyjJioPdM_r8fra2hf6pVOiTbYeXhOGzcqSHMEkxOCPbWgE_VwAFtMuNIF_gDuzI-GB5IBFTMKFsRJP7g8XXJzh0rxSSpAhajcR6DEHrtFhuh2dMfl5k1wv1XurxqP0EQ9peAKyOxdoPbYYOQ-kQzg3AM0fc8oIwbVAwtylooMAk-_qWz0k99HqtYy5waM7mcCaICbVoaJc9ePDpJlOQk-5UQ8LI_b-Dv2wZO6TpxcJd_PPpeRFTzd19SOxrgVuDE3m4VEO55d0l4YU1x7GPIlBa-46eGtdnOa1nlSDsmzhEyschGt48IC1IHd02fTuOZ5c_lKvo3O_El-K4EnEci4vnVFBu300iIhJmskTGuSytzbUQaIxBl6-CIsxXzSlUpLqm6ank0bpGUzL0UKudSsv01CM5z05M_NvN-0Mirkz7SlgOtpk4E2A3xTZ7YmnvcN6CaHnvcU000BD40X_nnOcivyYcO4PhTzpgInbaOHQ1SNBcTU4hgRjUVpkkjM-paXNQIz405_JlqitnAkQAtapk-eIymzxCq-GN "iot_provisioning_flow")
+
+## Porting the IoT Clients
+
+In order to port the clients to a target platform the following items are required:
+
+- Support for a C99 compiler.
+- Types such as `uint8_t` must be defined.
+- The target platform supports a stack of several kB (actual requirement depends on features being used and data sizes).
+- An MQTT over TLS client supporting QoS 0 and 1 messages.
+
+Optionally, the IoT services support MQTT tunneling over WebSocket Secure which allows bypassing firewalls where port 8883 is not open. Using WebSockets also allows usage of devices that must go through a WebProxy. Application developers are responsible with setting up the wss:// tunnel.
+
+## API
+
+### Connecting
+
+The application code is required to initialize the TLS and MQTT stacks.
+Two authentication schemes are currently supported: _X509 Client Certificate Authentication_ and _Shared Access Signature_ authentication.
+
+When X509 client authentication is used, the MQTT password field should be an empty string.
```

Comment:
nit: This is not necessarily true. In paho the password is set to NULL. So maybe a slight improvement here is:
```suggestion
When X509 client authentication is used, the MQTT password field should be empty.
```

Reply:
+1